addons/l10n_it_edi/models/account_edi_format.py

```diff
@@ -408,15 +408,64 @@ def _import_fattura_pa(self, tree, invoice):
+                    'section': 'seller',
+                    'vat_elements': tree.xpath('//CedentePrestatore//IdCodice'),
+                    'codice_fiscale_elements': tree.xpath('//CedentePrestatore//CodiceFiscale'),
+                },
+            }
+
+            default_move_type = self.env.context.get('default_move_type')
+            # The default_move_type is ...
+            #   * ... set when clicking on the "Upload" button on the invoices or bills views.
+            #     In that case we want to only import it as invoice or bill resepectively
+            #   * ... not set when importing from cron
+            if default_move_type is None:
+                # Try importing as incoming or outgoing move
+                incoming_possibilities = [True, False]
+            elif default_move_type in invoice.get_purchase_types(include_receipts=True):
+                # Only try importing as incoming move
+                incoming_possibilities = [True]
+            elif default_move_type in invoice.get_sale_types(include_receipts=True):
+                # Only try importing as outgoing move
+                incoming_possibilities = [False]
+            else:
+                _logger.warning("Cannot handle default_move_type '%s'.", default_move_type)
+                continue
+
+            # Consider the import via the "Upload" button on the invoices or bills view.
+            # There we only want to import moves for the currently allowed companies.
+            allowed_company_ids = self.env.context.get('allowed_company_ids')
+            allowed_company_domain = [('id', 'in', allowed_company_ids)] if allowed_company_ids else []
+
+            for incoming in incoming_possibilities:
+                info = buyer_seller_info[incoming]
+                vat_elements = info['vat_elements']
+                codice_fiscale_elements = info['codice_fiscale_elements']
+                company = vat_elements and self.env['res.company'].search([
+                    ('vat', 'ilike', vat_elements[0].text),
+                    *allowed_company_domain
+                ], limit=1)
                 if not company:
-                    # Only invoices with a correct VAT or Codice Fiscale can be imported
-                    _logger.warning('No company found with VAT or Codice Fiscale like %r.', elements[0].text)
-                    continue
+                    company = codice_fiscale_elements and self.env['res.company'].search([
+                        ('l10n_it_codice_fiscale', 'ilike', codice_fiscale_elements[0].text),
+                        *allowed_company_domain
+                    ], limit=1)
+                if company:
+                    break
+                else:
+                    _logger.warning('No (allowed) company found with VAT or Codice Fiscale from the %s section.', info['section'])
+
+            if not company:
+                _logger.warning('Could not determine company.')
+                continue
```

Comment:
I don't think we should log when we don't find the VAT in the section.

Use the `for..else` instead of an explicit `if`
https://docs.python.org/3/tutorial/controlflow.html#break-and-continue-statements-and-else-clauses-on-loops

Reply:
Thanks I did not know about this feature

Reply:
Credit goes to William, who pointed it out once on a review he did for me :)